bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-persistence.js

```diff
@@ -282,9 +306,12 @@ export default function defineOHBlocks_Persistence (f7, persistenceServices) {
     let code = ''
     const dayInfoSince = javascriptGenerator.valueToCode(block, 'dayInfoSince', javascriptGenerator.ORDER_NONE)
     const dayInfoUntil = javascriptGenerator.valueToCode(block, 'dayInfoUntil', javascriptGenerator.ORDER_NONE)
-    const dayInfo = dayInfoSince + ((dayInfoSince && dayInfoUntil) ? ' ,' : '') + dayInfoUntil
+    const dayInfo = dayInfoSince + ((dayInfoSince && dayInfoUntil) ? ', ' : '') + dayInfoUntil
     let skipPrevious = javascriptGenerator.valueToCode(block, 'skipPrevious', javascriptGenerator.ORDER_NONE)
-    skipPrevious = ((skipPrevious === 'undefined') ? false : skipPrevious)
+    skipPrevious = (skipPrevious === 'undefined') ? false : skipPrevious
+
+    let riemannType = block.getFieldValue('riemannType')
+    riemannType = (riemannType === 'undefined') ? '' : `, ${itemCode}.persistence.${riemannType}`
```

Comment:
Let's use static access to the RiemannType:

```suggestion
    riemannType = (riemannType === 'undefined') ? '' : `, items.${riemannType}`
```

Reply:
done